### PR TITLE
Allow for multiple ASes

### PR DESCRIPTION
**Pull request by leonerd**
Changes the storage of AS instance-related variables into arrays. Though currently these are 1-element arrays.

### CODE REVIEW

File: tests/02as-info.pl [hunk 1]
```diff
@@ -8,19 +8,25 @@ sub gen_token
...
+
+         my $localpart = "as-user-$idx";
+
+         Future->done( ASInfo(
+            $localpart,
+            "\@${localpart}:localhost:${port}",
+            gen_token( 32 ),
+            gen_token( 32 ),
+            "/appservs/$idx",
+            $idx,
```
Comment (illicitonion) [inline tests/02as-info.pl]: stringify

File: tests/05synapse.pl [hunk 2]
```diff
@@ -79,24 +79,32 @@ END
...
-               }
-            } );
+            my @confs;
+
+            foreach my $idx ( 0 .. $#as_infos ) {
+               my $as_info = $as_infos[$idx];
+
+               my $appserv_conf = $synapse->write_yaml_file( "appserv-$idx.yaml", {
+                  url      => $test_server_info->client_location . $as_info->path,
+                  as_token => $as_info->as2hs_token,
```
Comment (illicitonion) [inline tests/05synapse.pl]: Set ID in the yaml config

**Comment by illicitonion** [comment]
LGTM with minor comments

File: tests/05synapse.pl (hunk 2 above)
Comment (leonerd) [inline tests/05synapse.pl]: So, looking at the specification for the application service YAML config, I don't see the ID field documented:

  http://matrix.org/docs/spec/r0.0.1/application_service.html#registration

Comment (leonerd) [inline tests/05synapse.pl]: Ah, but it is in the source code for that file. How odd. Maybe the site hasn't been pushed lately.

File: tests/02as-info.pl (hunk 1 above)
Comment (leonerd) [inline tests/02as-info.pl]: Done d41d64b

File: tests/05synapse.pl (hunk 2 above)
Comment (leonerd) [inline tests/05synapse.pl]: Done d41d64b

**Comment by leonerd** [comment]
PTAL

**Comment by illicitonion** [comment]
LGTM

**Pull request merged**